benchmark/rclnodejs/client-stress-test.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2018 Intel Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+/* eslint-disable camelcase */
+const rclnodejs = require('../../index.js');
+
+rclnodejs.init().then(() => {
+  console.log('The client will send a GetMap request(response contains a size of 10MB array) every 100ms.');
+  console.log(`Begin at ${new Date().toString()} and may end in 1 hour.`);
+
+  const node = rclnodejs.createNode('stress_client_rclnodejs');
+  const client = node.createClient('nav_msgs/srv/GetMap', 'get_map');
+  let sentTimes = 0;
+  let receivedTimes = 0;
+  let totalTimes = 36000;
+
+  let timer = setInterval(() => {
+    if (sentTimes++ > totalTimes) {
+      clearInterval(timer);
+    } else {
+      client.sendRequest({_dummy: true}, (response) => {
+        if (++receivedTimes > totalTimes) {
```

Comment:
Is there a case that the `response` may be lost in some low performance device? Thus the `receivedTimes` could be never greater than `totalTimes`, i.e. the following code will never be called.

Reply:
If the network is broken, the client will not receive the response any more. 
If the device is low performance and the connection is still alive, I think the client will receive the response as the DDS is over TCP connection in ROS 2.

How about ignoring the count of sending the request and counting the response received only?

```JavaScript
 let timer = setInterval(() => {
      client.sendRequest({_dummy: true}, (response) => {
        if (++receivedTimes > totalTimes) {
          clearInterval(timer);
          rclnodejs.shutdown();
          console.log(`End at ${new Date().toString()}`);
        }
      });
}, 100);
```